app/src/main/java/ch/hikemate/app/utils/MapUtils.kt

```diff
@@ -0,0 +1,109 @@
+package ch.hikemate.app.utils
+
+import android.util.Log
+import ch.hikemate.app.model.route.Bounds
+import kotlin.math.cos
+import org.osmdroid.util.GeoPoint
+
+/**
+ * Calculates the geographical center of a given bounding box. This function computes the center
+ * latitude as the average of the minimum and maximum latitudes, and the center longitude as the
+ * average of the minimum and maximum longitudes. It also handles edge cases where the longitude
+ * crosses the Date Line (±180 degrees), adjusting the center longitude accordingly to ensure it
+ * stays within the valid range of -180 to 180 degrees.
+ *
+ * @param bounds The bounding box (minLat, maxLat, minLon, maxLon) defining the area for which to
+ *   calculate the center.
+ * @return A `GeoPoint` representing the geographical center of the provided bounds.
+ */
+fun getGeographicalCenter(bounds: Bounds): GeoPoint {
+  Log.d(
+      "Bounds",
+      "minLat: ${bounds.minLat}, maxLat: ${bounds.maxLat}, minLon: ${bounds.minLon}, maxLon: ${bounds.maxLon}")
+  val minLong = bounds.minLon
+  val maxLong = bounds.maxLon
+  val minLat = bounds.minLat
+  val maxLat = bounds.maxLat
+
+  val centerLat = (minLat + maxLat) / 2
+  var centerLong = (minLong + maxLong) / 2
+
+  // Adjust if the longitude crosses the Date Line (i.e., difference > 180 degrees)
+  if (maxLong - minLong > 180) {
+    centerLong = (minLong + maxLong + 360) / 2
+  } else if (minLong - maxLong > 180) {
+    centerLong = (minLong + maxLong - 360) / 2
+  }
+
+  // Normalize longitude to be in the range -180 to 180
+  if (centerLong > 180) {
+    centerLong -= 360
+  } else if (centerLong < -180) {
+    centerLong += 360
+  }
```

Comment:
Thanks for the good question, it took me some wrapping my head around too to catch these edge cases. 
_(Btw, the code you're looking at is slightly outdated (I must have not pushed some small changes I made, I used to assume that maxLong/Lat can also be smaller than minLong/Lat), but the ideas stay the same.)_

The coordinates used by the hike route API use latitude and longitude values. Imagine now, that you have a hike that crosses from Russia (~-170° to Alaska ~+170°). The formula without this check would calculate the center like this:
`(-170 + 170)/2 = 0°`
which is the meridian crossing through Greenwich UK, and nowhere between Russia and Alaska. 
This check would notice that `170 - (-170) > 180`, and adjust the center longitude to be instead:
 `(-170 + 170 + 360)/2 = 180° `
which is correct

For the second check, after having looked at it again with a clear head and looking at the [OSM documentation](https://wiki.openstreetmap.org/wiki/180th_meridian), seems to be unnecessary. I thought, or rather ChatGPT told me, that map systems can't handle a longitude of exactly 180°, since they tend to use -180°. These two longitudes overlap, so this might cause issues, but seemingly OSM can handle this edge case without issues, so I deleted the second check.

Reply:
Thanks for the explanation !